pennylane_lightning_gpu/src/algorithms/ObservablesGPUMPI.hpp

```diff
@@ -0,0 +1,403 @@
+// Copyright 2022-2023 Xanadu Quantum Technologies Inc. and contributors.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <functional>
+#include <vector>
+
+#include "StateVectorCudaMPI.hpp"
+
+namespace Pennylane::Algorithms {
+
+/**
+ * @brief A base class for all observable classes.
+ *
+ * We note that all subclasses must be immutable (does not provide any setter).
+ *
+ * @tparam T Floating point type
+ */
+
+template <typename T>
+class ObservableGPUMPI
```

Comment:
We could probably just import the LightningGPU observable based class here and it should work the same, right? Or would that require some templating of the SV type?